src/data-list/data-list.ts

```diff
@@ -0,0 +1,277 @@
+namespace ddui {
+    export interface ListRow {
+        $selected?: boolean;
+    }
+
+    export interface ListConfig {
+        url: string;
+        id?: string;
+        paging?: boolean;
+        responseListName?: string;
+        responseCountName?: string;
+    }
+
+    export class DataList<T extends ListRow> {
+
+        id: string;
+        rows: T[];
+        selectedRows: T[];
+        url: string;
+        count: number;
+        data: any;
+        filter: any;
+        paging: boolean;
+        selectedAllPages: boolean;
+        isLoading: boolean;
+
+        private responseListName: string;
+        private responseCountName: string;
+        private onListResponseSuccess: any;
+        private onListResponseError: any;
+        private initFilterFunc: Function;
+
+        constructor(config: ListConfig, private $http: ng.IHttpService, private $location: ng.ILocationService) {
+            this.id = config.id;
+            this.rows = [];
+            this.selectedRows = [];
+            this.count = 0;
+            this.data = null;   // will be assigned raw last response
+            this.filter = { skip: 0, limit: 25 };
+            this.paging = config.paging;
+            this.selectedAllPages = false;
+            this.isLoading = false;
+            this.url = config.url;
+            this.responseListName = config.responseListName || 'items';
+            this.responseCountName = config.responseCountName || 'count';
+            this.initFilterFunc = () => { return {}; };
+        }
+
+        onSuccess(callback: (list: T[], count: number) => void) {
+            this.onListResponseSuccess = callback;
+        }
+
+        onError(callback: (data: any) => void) {
+            this.onListResponseError = callback;
+        }
+
+        setFilter(filterFunc) {
+            this.initFilterFunc = filterFunc;
+            var filter = this.initFilterFunc();
+            if (typeof (filter) !== 'object') {
+                throw new Error('initFilterFunc should return object with filter properties');
+            }
+
+            angular.extend(this.filter, filter);
+
+            this.loadLocationParams(this.filter);
+        }
+
+        submitFilter() {
+            this.setLocationParams(this.filter);
+            return this.updateList();
+        }
+
+        resetFilter() {
+            this.filter = this.initFilterFunc();
+            this.setLocationParams(this.filter);
+            this.updateList();
+        }
+
+        updateList() {
+            this.isLoading = true;
```

Comment:
I think it would be nice to have possibility to refresh all list. Now if you want to refresh all list, you have to set your filter.skip to 0 from controller and after update set to the value it was before to do that.